include/nuttx/spinlock.h

```diff
@@ -449,4 +454,120 @@ void spin_unlock_irqrestore_wo_note(FAR spinlock_t *lock, irqstate_t flags);
 #  define spin_unlock_irqrestore_wo_note(l, f) up_irq_restore(f)
 #endif
 
+#if defined(CONFIG_RW_SPINLOCK)
+
+/****************************************************************************
+ * Name: check_bit
+ *
+ * Description:
+ *  Check offset is set or not.
+ *
+ * Input Parameters:
+ *   lock - A lock instance.
+ *   offset - from right to left.
+ *
+ * Returned Value:
+ *   bool
+ *
+ * Assumptions:
+ *   No.
+ ****************************************************************************/
+
+inline bool check_bit(spinlock_t lock, uint8_t offset)
```

Comment:
Can this be a macro?

Reply:
This function is not needed so I deleted it.